research/train.py

```diff
@@ -0,0 +1,42 @@
+import argparse
+import torch
+import json
+from time import time
+from steganogan import SteganoGAN
+from steganogan.loader import DataLoader
+from steganogan.critics import BasicCritic
+from steganogan.decoders import DenseDecoder
+from steganogan.encoders import BasicEncoder, ResidualEncoder, DenseEncoder
+
+torch.manual_seed(42)
+timestamp = int(time())
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--epochs', default=32, type=int)
+parser.add_argument('--encoder', default="basic", type=str)
+parser.add_argument('--data_depth', default=1, type=int)
+parser.add_argument('--hidden_size', default=32, type=int)
+parser.add_argument('--dataset', default="div2k", type=str)
+args = parser.parse_args()
```

Comment:
Added output option which will (optionally) save a copy of the model to the user's chosen location. I didn't add the decoder option because we do not use the BasicDecoder at all (and we should pretty much never use it), I think it was an artifact of the refactoring from the original research repo.

If / when we add a viable alternative decoder, I agree we should add an option. Until then, I would say keep it simple for now.

Reply:
Sounds good!